src/services/swap_service.ts

```diff
@@ -479,10 +492,13 @@ export class SwapService {
                 assetSwapperOpts,
             );
         } else if (buyAmount !== undefined) {
+            const buyAmountScaled = buyAmount
+                .times(affiliateFee.buyTokenPercentageFee + 1)
+                .integerValue(BigNumber.ROUND_DOWN);
```

Comment:
I feel like this should be `ROUND_UP`.

Reply:
I believe the rounding mode needs to be the same between this and `getAffiliateFeeAmounts`

Reply:
Isn't it possible we'll be under by 1 wei after the affiliate fee is taken out by the transformer?

Reply:
I tried to work out the math but gave up
```
    for (let i = 0; i < 100000; i++) {
        it(`Test ${i}`, () => {
            const buyAmount = getRandomInteger(0, constants.ONE_ETHER);
            const buyTokenPercentageFee = getRandomFloat(0, 1);
            const buyAmountScaled = buyAmount.times(buyTokenPercentageFee.plus(1)).integerValue(BigNumber.ROUND_DOWN);
            const buyTokenFeeAmount = buyAmountScaled
                .times(buyTokenPercentageFee)
                .dividedBy(buyTokenPercentageFee.plus(1))
                .integerValue(BigNumber.ROUND_DOWN);
            expect(buyAmountScaled.minus(buyTokenFeeAmount)).to.bignumber.equal(buyAmount);
        });
    }
```
> `100000 passing (26s)`